### ISSUE TITLE
Merge album views of separate discs into one

### ISSUE THREAD

**Issue by jojosoft** [Rello/audioplayer#522]
Big thanks to @Rello for actively developing my favorite Nextcloud app! I was happy to dump iTunes and its weird (artificial) restrictions – it's like hosting your own free and privacy-respecting Spotify. However, over the last year, I noticed some things that got a little worse than before (mainly related UI), so I decided to open several smaller issues to address them.

### Actual behavior
When there are albums in the collection that have multiple discs, they are listed as separate albums in the album cover view. But since the format of the title numbers already includes the disc numbers (for example 1-1, 1-2 ... 1-n for the first disc and 2-1, 2-2 ... 2-n for the second disc), it would make sense to only display the album once and then list all titles of all discs inside one tracklist view. At least personally, I would like this much more. My idea is that you could maybe display the tracks as usual and separate the different discs with a horizontal line or something similar. This would be perfect. Since we need (?) to account for almost arbitrarily long tracklists anyway, I hope that the list's length would not be a problem.

### Steps to reproduce the behavior
1. Add music files to the collection which belong to the same album, but have different disc numbers
2. See how in album view, one album is displayed for each disc

### Additional information (optional)
Even if this aspect purely depends on preference, I would still really appreciate a checkbox for this in the settings!

### Server configuration
**Operating system:**
Debian 10

**Web server:**
Apache

**Database:**
mysql

**Nextcloud/ownCloud version:** (see `/status.php`)
19.0.6

**Audio Player version:** (see `/settings/apps`)
2.13.1

**Are you using external storage? If yes, which one:** `local/smb/sftp/...`
No.

### Client configuration
**Browser:**
Firefox 84.0

**Comment by Rello** [comment]
Hello,

no setting for a display feature like that :-)
actually it should be as you describe. can you confirm that the 2 discs actually have the same name&artist?
i will double check here - but I am using exactly this

**Comment by jojosoft** [comment]
Yes, I made sure that the meta tags are the same. I use Kid3 to prepare all my MP3s for Audio Player.
![8](https://user-images.githubusercontent.com/9149359/103798384-e7fec400-5049-11eb-8728-f0f0a1f9f566.png) ![7](https://user-images.githubusercontent.com/9149359/103798385-e8975a80-5049-11eb-8cdb-b523741b4675.png)

**Comment by Rello** [comment]
Hello,
its about this setting. The scanner is only putting everything in one album, if the folder of the file is the same.
https://github.com/Rello/audioplayer/blob/master/lib/Controller/ScannerController.php#L304

The reason was, that there could be several albums for "Greatest Hits" e.g.
It is also required if separate disks have different covers.
they where grouped into one album then with a great mixup. 
https://github.com/Rello/audioplayer/issues/271

your case is having the titles in 2 folders?
could they be joined?

**Comment by jojosoft** [comment]
Oh, I see what's the problem here. I just didn't assume that there will be two albums with equal artists **and** equal name that the user wouldn't like to have inside one list. But you're right, I currently have subfolders for each disk, I didn't know the scanner is actually taking folder structure into account.

Unfortunately, I cannot test it right now (without greater efforts) because Audio Player doesn't seem to be able to remove an album from the collection once it has been added. I tried temporarily moving the audio files in question to the trash bin, but it still shows them. (But isn't able to load and play them anymore.) If I use the "reset library" button, I will probably lose all of the star assignments..? I could probably take some time to manually modify the database, but I'm not that experienced with it...

**Comment by jojosoft** [comment]
I just tried something and it worked! Multiple CDs of one album need to go into the same folder, as you explained.
In the Nextcloud data folder, I executed: `touch johannes/files/path/to/album/*`
And in the Nextcloud system folder: `sudo -u www-data php occ files:scan -p "johannes/files/path/to/album/"`
This actually makes Audio Player reload just the audio files for this album, this time inside a single album.
